source/common/common/logger.h

```diff
@@ -123,7 +124,66 @@ template <Id id> class Loggable {
 
 } // Logger
 
-#ifdef NDEBUG
+/**
+ * Base logging macros.  It is expected that users will use the convenience macros below rather than
```

Comment:
Now that we have macros everywhere, do you want to add file/line info also? Should be easy.